common_package_py/commands.py

```diff
@@ -0,0 +1,344 @@
+            Parses and checks a JSON string based on the given definition.
+
+        parse_check_json(json_obj: json, definition: dict) -> json:
+            Parses and checks a JSON object based on the given definition.
+    """
+
+    def __init__(self) -> None:
+        pass
+
+    @staticmethod
+    def parse_check_json_str(json_str: str, definition: dict) -> json:
+        """
+        Parses and checks a JSON string based on the given definition.
+
+        Note:
+            The function can only be fully used on shallow JSONs. Arrays or
+            encapsulations contents' will not be checked! Call this function several
+            times with the different parts to get that behavior.
+
+        Args:
+            json_str (str): The JSON string to parse and check.
+            definition (dict): The definition of the JSON object structure.
+
+        Returns:
+            json: The parsed and checked JSON object.
+
+        Raises:
+            RuntimeError: With an error message why the parsing or check failed
+        """
+
+        try:
+            candidate = json.loads(json_str)
+        except json.JSONDecodeError as e:
+            raise RuntimeError(
+                f"CommandDefinitions::parse_check_json_str: Failed to parse JSON: {str(e)}, lineno: {e.lineno}, colno: {e.colno}, pos: {e.pos}")
+
+        return CommandDefinitions.parse_check_json(candidate, definition)
+
+    @staticmethod
+    def parse_check_json(json_obj: json, definition: dict) -> json:
+        """
+        Parses and checks a JSON object based on the given definition.
+
+        Only use this version if you already have a json object.
+        If you only have a string, use the `parse_check_json_str` function
+        instead to do a proper parsing.
+
+        Steps:
+        - Checks that no undefined keys are in the JSON
+        - Checks that all required keys exist
+        - Checks that all values have the correct type
+
+        Note:
+            The function can only be fully used on shallow JSONs. Arrays or
+            encapsulations contents' will not be checked! Call this function several
+            times with the different parts to get that behavior.
+
+        Args:
+            json_obj (json): The JSON object to parse and check.
+            definition (dict): The definition of the JSON object structure.
+
+        Returns:
+            json: The parsed and checked JSON object.
+
+        Raises:
+            RuntimeError: If unknown keys are found, if required keys are missing, if values are out of bounds,
+                or if values have the wrong type. An error message why the check failed is included.
+        """
+
+        # Check that all keys are allowed
+        # If true, unknown keys were found and an exception will be thrown
+        unknown_key_found = False
+        unknown_keys = []  # List with all unknown keys that were found
+
+        for key, val in json_obj.items():
```

Comment:
Set logic verwenden:
```suggestion
        unknown_keys = json_obj.keys() - definition
        if unkown_keys:
            pass # Error
```

Reply:
Hab ich gemacht